frontend/shared/components/inputs/Slider.vue

```diff
@@ -0,0 +1,334 @@
+<template>
+    <div class="slider-box">
+        <div class="slider-container" @mousedown="dragStart" @touchstart="dragStart">
+            <div class="slider" ref="slider">
+                <div class="middle">
+                    <div class="fill" :style="{width: this.handlePercentage+'%'}" :class="{animate: animate}"></div>
+                </div>
+                <div class="handle" ref="handle" :style="{left: this.handlePercentage+'%'}" :class="{animate: animate}"></div>
+            </div>
+        </div>
+       
+        <div class="number">
+            <input type="number" pattern="[0-9]*" v-model.number="value" v-on:change="updateSlider" @focus="delayedSelect">
+        </div>
+    </div>
+</template>
+
+<script lang="ts">
+import { Component, Prop, Vue } from 'vue-property-decorator';
+
+@Component
+export default class Slider extends Vue {
+    min: number = 0;
+    max: number = 300;
+
+    /// Offset from the center of the handle during drag.
+    startOffset: number = 0;
+    handlePercentage: number = 50;
+    round?: number = 10;
+
+    // if softBounds = true, maximum and minimum is overrideable by the user 
+    // when he manually enters a number. This is usefull if the maximum and 
+    // minimum is just a convenience.
+    softBounds: boolean = true;
+    animate: boolean = false;
+
+    value: number = 150;
+    dragging: boolean = false;
+
+    mounted() {
+        
+    }
+
+    attach() {
+        document.addEventListener('mousemove', this.mouseMove, {passive: false});
+        document.addEventListener('touchmove', this.mouseMove, {passive: false});
+
+        document.addEventListener('mouseup', this.mouseUp, {passive: false});
+        document.addEventListener('touchend', this.mouseUp, {passive: false});
+    }
+
+    detach() {
+        document.removeEventListener('mousemove', this.mouseMove);
+        document.removeEventListener('touchmove', this.mouseMove);
+
+        document.removeEventListener('mouseup', this.mouseUp);
+        document.removeEventListener('touchend', this.mouseUp);
+    }
+
+    delayedSelect(event) {
+        // Prevent default seems to be required because else it wont work 100% of the time
+        event.preventDefault();
+
+        // Select all the text
+        event.target.select();
+
+        const handler = (event) => {
+            // Safari fix
```

Comment:
More info?

Reply:
Safari deselects the text on mouse up. This fix works most of the time.
```suggestion
            // Safari deselects the text on mouse up, we need to prevent this
```